tests/rest/admin/test_user.py

```diff
@@ -1165,6 +1168,55 @@ def test_get_rooms(self):
         self.assertEqual(number_rooms, channel.json_body["total"])
         self.assertEqual(number_rooms, len(channel.json_body["joined_rooms"]))
 
+    @defer.inlineCallbacks
+    def test_get_rooms_with_nonlocal_user(self):
+        """
+        Tests that a normal lookup for rooms is successfully
+        """
+        # Create rooms and join
+        other_user_tok = self.login("user", "pass")
+        total_number_rooms = 10
+        actually_joined_rooms = total_number_rooms // 2
+        rooms = []
+        for n in range(total_number_rooms):
+            rooms.append(
+                self.helper.create_room_as(self.other_user, tok=other_user_tok)
+            )
+
+        # Add a remote user to a few of these rooms
+        event_builder_factory = self.hs.get_event_builder_factory()
+        event_creation_handler = self.hs.get_event_creation_handler()
+        storage = self.hs.get_storage()
+        for n in range(actually_joined_rooms):
+            builder = event_builder_factory.for_room_version(
+                RoomVersions.V1,
+                {
+                    "type": "m.room.member",
+                    "sender": "@joiner:remote_hs",
+                    "state_key": "@joiner:remote_hs",
+                    "room_id": rooms[n],
+                    "content": {"membership": "join"},
+                },
+            )
+
+            event, context = yield defer.ensureDeferred(
+                event_creation_handler.create_new_client_event(builder)
+            )
+
+            yield defer.ensureDeferred(
+                storage.persistence.persist_event(event, context)
+            )
```

Comment:
You can remove the `defer.inlineCallbacks` decorator and make the following change to get a better error message:

```suggestion
            event, context = self.get_success(
                event_creation_handler.create_new_client_event(builder)
            )

            self.get_success(
                storage.persistence.persist_event(event, context)
            )
```